src/Components/WebAssembly/Build/src/targets/Blazor.MonoRuntime.targets

```diff
@@ -318,12 +320,43 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="_GenerateBlazorBootJsonInputHash">
+    <ItemGroup>
+      <_BlazorBootJsonHashInput Include="@(IntermediateAssembly)" />
+      <_BlazorBootJsonHashInput Include="@(_BlazorOutputWithTargetPath)" />
+      <_BlazorBootJsonHashInput Include="@(_BlazorConfigFile)" />
+    </ItemGroup>
```

Comment:
Could you help me understand why `_GenerateBlazorBootJsonInputHash` does this?

Since its result comes from the `<Hash>` task, it's purely a function of a list of filenames in `@(IntermediateAssembly)`, `@(_BlazorOutputWithTargetPath)`, and `@(_BlazorConfigFile)`. Its output does not vary according to those files' contents or timestamps.

When would we expect this list of filenames to change?

I would have thought we need `_GenerateBlazorBootJson` to re-rerun when the modify-timestamps on the `appsettings.*.json` files change, not just their filenames.

Or am I misunderstanding how this works?

Reply:
The hash trick is useful when you want to re-run if the list of file inputs change, without the timestamp having changed. Think of a scenario where you start with `appsettings.json` and `appsettings.Development.json`, but then go on to delete the second file between runs. All the timestamps are still accurate i.e. `timestamp(boot.json) > timestamp(all the inputs)`, but you still want to regenerate it. 

Hashing the inputs and writing it to disk only when the hash changes helps keep track of cases like this.

Reply:
Makes sense. Thanks!